group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1815,18 +1814,18 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, CoordinatorRecord>
                 numMembers
             );
 
-            if (!subscriptionMetadata.equals(group.subscriptionMetadata())) {
+            long subscriptionMetadataHash = Utils.hashSubscriptionMetadata(subscriptionMetadata);
+            if (subscriptionMetadataHash != group.subscribedTopicMetadataHash()) {
```

Comment:
Should we align the naming? `subscriptionMetadataHash` or `subscribedTopicMetadataHash`